azurelinuxagent/ga/exthandlers.py

```diff
@@ -978,33 +978,40 @@ def write_ext_handlers_status_to_info_file(self, vm_status):
             "python_version": "Python: {0}.{1}.{2}".format(PY_VERSION_MAJOR, PY_VERSION_MINOR, PY_VERSION_MICRO),
             "crp_supported_features": [name for name, _ in get_agent_supported_features_list_for_crp().items()],
             "extension_supported_features": [name for name, _ in
-                                             get_agent_supported_features_list_for_extensions().items()]
+                                             get_agent_supported_features_list_for_extensions().items()],
+            "last_successful_status_upload_time": time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime())
         }
 
         if vm_status is not None:
             # Convert VMStatus class to Dict.
             data = get_properties(vm_status)
-            goal_state_status = data['vmAgent']['vm_artifacts_aggregate_status']['goal_state_aggregate_status']
 
-            agent_details["daemon_version"] = data['vmAgent']['version']
-            agent_details["distro_details"] = "{0}:{1}".format(data['vmAgent']['osname'], data['vmAgent']['osversion'])
-            agent_details["agent_status"] = data['vmAgent']['status']
-            agent_details["agent_message"] = data['vmAgent']['message']
-            agent_details["agent_hostname"] = data['vmAgent']['hostname']
+            # .get handles the keyError and accepts a default value as 2nd parameter. If no second parameter is
+            # provided it uses None. Using {'default': None} in the case of 'vmAgent' or any other key is not available
+
+            agent_details["daemon_version"] = data.get('vmAgent', {'default': None}).get('version')
```

Comment:
In case 'vmAgent' is not there, the default would return None. 
If the value is None, then 'None'.get('version') would throw too right?

Why not have an if condition at the top and go like - 

```python
if 'vmAgent' in data:
    # Fetch other stuff
```

And maybe do this for all nested properties?

Reply:
I found . get to be a good way to handle nested dict. In case of VmAgent not available it will consider vmAgent = {'default': None} , which is a valid JSON and will not throw KeyError. I.e it will try to get 'version' from {'default': None} and since 'version' is not available, it will return None but not Key Error.

The problem with nested Ifs is for every property we have to surround it by an If , but felt like .get handles it more elegantly.